src/Datadog.Trace.AspNet/TracingHttpModule.cs

```diff
@@ -44,11 +47,18 @@ public TracingHttpModule(string operationName)
         /// <inheritdoc />
         public void Init(HttpApplication httpApplication)
         {
-            _httpApplication = httpApplication;
-
-            // The first HttpModule to run Init for this HttpApplication will register for events
+            // Intent: The first HttpModule to run Init for this HttpApplication will register for events
+            // Actual: Each HttpApplication that comes through here is potentially a new .NET object, even
+            //         if it refers to the same web application. Based on my reading, it appears that initialization
+            //         is done for several types of resources. Read more in this SO article -- look at Sunday Ironfoot's
```

Comment:
You can link directly to Ironfoots answer: https://stackoverflow.com/a/2416546/24231